diff --git a/app/pre_process.py b/app/pre_process.py
index 2e32605..943647c 100644
--- a/app/pre_process.py
+++ b/app/pre_process.py
@@ -45,7 +45,7 @@ def process_video(video_file_name, socketio):
     update_user_video_data(video_file_name, None, None, False, True)
     while step_seconds < video_length_seconds:
         seconds_since_start = time.time() - start_time
-        print(f"{seconds_to_timestamp(step_seconds)}/{seconds_to_timestamp(video_length_seconds)}  Processing for: {seconds_to_timestamp(seconds_since_start)}")
+        print(f"{seconds_to_timestamp(step_seconds)}/{seconds_to_timestamp(video_length_seconds)}  Processing for: {seconds_to_timestamp(int(seconds_since_start))}")
         cap.set(cv2.CAP_PROP_POS_FRAMES, step_seconds * video_fps)
         text = run_ocr(*cap.read())
         response = LlamaInterface.query_with_default(text)
diff --git a/app/remotellama.py b/app/remotellama.py
index 77da355..b90864c 100644
--- a/app/remotellama.py
+++ b/app/remotellama.py
@@ -7,7 +7,10 @@ class LlamaInterface:
     headers = {"Content-Type": "application/json"}
     def query(question):
         data = {"prompt": question}
-        response = requests.post(LlamaInterface.url + 'llama', headers=LlamaInterface.headers, json=data)
+        try:
+            response = requests.post(LlamaInterface.url + 'llama', headers=LlamaInterface.headers, json=data)
+        except requests.exceptions.ConnectTimeout:
+            return "Timeout Error Has Occurred"
         response_data = response.json()
         # Check if the response is a dictionary before trying to access it
         if isinstance(response_data, dict):
